src/main/fox_demo.c

```diff
@@ -0,0 +1,3005 @@
+        }
+        if (gLaserStrength[gPlayerNum] != 0) {
+            gLaserStrength[gPlayerNum] = 0;
+            func_8004D3D4(arg0, 0x142, &gItems[0]);
+        }
+    }
+}
+
+void func_8004D738(Player* arg0) {
+    arg0->pos.y += 30.0f;
+    func_8007D0E0(arg0->pos.x, arg0->pos.y, arg0->unk_138, 6.0f);
+    if (gCamCount == 1) {
+        func_8007BFFC(arg0->pos.x, arg0->pos.y, arg0->unk_138, 0.0f, 0.0f, 0.0f, 3.0f, 0x50);
+        func_8007C688(arg0->pos.x, arg0->pos.y, arg0->unk_138, 3.0f, 0x320);
+    } else {
+        func_8007BFFC(arg0->pos.x, arg0->pos.y, arg0->unk_138, 0.0f, 0.0f, 0.0f, 3.0f, 0xA);
+    }
+    func_8004D440(arg0);
+}
+
+void func_8004D828(Player* arg0) {
+    s32 i;
+    Vec3f src;
+    Vec3f dest;
+
+    Math_SmoothStepToF(&arg0->unk_0E4, 0.0f, 0.1f, 1.0f, 0.01f);
+    arg0->pos.x += arg0->vel.x;
+    arg0->pos.y += arg0->vel.y;
+    if ((gCurrentLevel != 6 || D_80178284 == 0) && (gLevelType == 0 || gCurrentLevel == 0x11)) {
+        arg0->vel.y -= 0.5f;
+        arg0->unk_0E4 -= 2.0f;
+    }
+    arg0->pos.z += arg0->vel.z;
+    arg0->unk_138 = arg0->pos.z;
+    arg0->unk_0F8 = (arg0->unk_0EC + arg0->unk_12C) + arg0->unk_130;
+    arg0->unk_12C += 15.0f;
+    if (arg0->unk_1D0 != 0) {
+        arg0->unk_0E8 += 11.0f;
+        arg0->unk_0E4 += 17.0f;
+    }
+    if (gCamCount == 1) {
+        if (!(gFrameCount & 1)) {
+            func_8007D24C((Rand_ZeroOne() - 0.5f) * 20.0 + arg0->pos.x, (Rand_ZeroOne() - 0.5f) * 20.0 + arg0->pos.y,
+                          arg0->unk_138, 2.2f);
+        }
+    } else if (!(gFrameCount & 3)) {
+        func_8007D10C(((Rand_ZeroOne() - 0.5f) * 10.0f) + arg0->pos.x, (Rand_ZeroOne() - 0.5f) * 10.0f + arg0->pos.y,
+                      ((Rand_ZeroOne() - 0.5f) * 10.0f) + arg0->unk_138, 2.2f);
+    }
+    if ((arg0->pos.y < arg0->unk_0A4) && (arg0->unk_1D0 == 0)) {
+        arg0->pos.y = arg0->unk_0A4;
+        arg0->unk_284 = 0;
+        arg0->timer_220 = 0;
+        arg0->vel.y = 10.0f;
+        arg0->unk_1D0 = 1;
+        func_800A6070(&arg0->unk_460, 0x29000000U);
+        if ((gCurrentLevel == 0) || (gCurrentLevel == 0xE)) {
+            func_80062C38(arg0->pos.x, arg0->pos.z);
+        } else {
+            func_8007D0E0(arg0->pos.x, arg0->pos.y, arg0->unk_138, 3.0f);
+        }
+        if (arg0->wings.rightState == 2) {
+            func_800A69F8(1, arg0->hit1.x, arg0->hit1.y, arg0->hit1.z);
+            arg0->wings.rightState = 1;
+            func_8007D0E0(arg0->hit1.x, arg0->hit1.y, arg0->hit1.z, 2.0f);
+        }
+        if (arg0->wings.leftState == 2) {
+            func_800A69F8(0, arg0->hit2.x, arg0->hit2.y, arg0->hit2.z);
+            arg0->wings.leftState = 1;
+            func_8007D0E0(arg0->hit2.x, arg0->hit2.y, arg0->hit2.z, 2.0f);
+        }
+    } else if ((((arg0->timer_220 > 0 || arg0->pos.y < arg0->unk_0A4) || arg0->pos.y < D_80177CC0) ||
+                arg0->timer_1FC == 0) &&
+               arg0->timer_1F8 == 0) {
```

Comment:
```suggestion
        } else if (((arg0->timer_220 > 0) || (arg0->pos.y < arg0->unk_0A4) || (arg0->pos.y < D_80177CC0) ||
                (arg0->timer_1FC == 0)) &&
               (arg0->timer_1F8 == 0)) {
```
We prefer parentheses around clauses of conditionals to make the order of operations clear.